### DISCUSSION TITLE
Extracting the heat source without running the thermal model

### DISCUSSION THREAD

**Discussion by danielwardstfc** [pybamm-team/PyBaMM#4207] [Q&A] [answered]
Hi there, I'm running an electrochemical model (SPM) for now and am looking to calculate the heat source Q described in the thermal lumped model. I don't need to calculate the temperature flow through the battery. Is there a way I can calculate Q without running the thermal model, i.e. just by calculating Q from the outputs of the SPM. 

Thanks very much.

Daniel

**Comment by valentinsulzer** [comment, marked as answer]
There is an option "calculate heat source for isothermal models" which you can set to "true", then variables like "Total heating [W]" will be non-zero

**Reply by danielwardstfc** [reply]
Perfect, that does it.  Thank you very much
